src/Components/Web.Extensions/src/wwwroot/headManager.js

```diff
@@ -0,0 +1,110 @@
+(function () {
+    // Local helpers
+
+    const blazorIdAttributeName = '_blazor_id';
+    const headCommentRegularExpression = /\W*Head:[^{]*(.*)$/;
+    const prerenderedTags = [];
+
+    function createHeadTag({ tagName, attributes }, id) {
+        const tagElement = document.createElement(tagName);
+
+        if (id) {
+            tagElement.setAttribute(blazorIdAttributeName, id);
+        }
+
+        if (attributes) {
+            for (const key in attributes) {
+                if (attributes.hasOwnProperty(key)) {
+                    tagElement.setAttribute(key, attributes[key]);
+                }
+            }
+        }
+
+        document.head.appendChild(tagElement);
+
+        return tagElement;
+    }
+
+    function resolvePrerenderedHeadComponents(node) {
+        node.childNodes.forEach((childNode) => {
+            const headElement = parseHeadComment(childNode);
+
+            if (headElement) {
+                applyPrerenderedHeadComponent(headElement);
+            } else {
+                resolvePrerenderedHeadComponents(childNode);
+            }
+        });
+    }
+
+    function applyPrerenderedHeadComponent(headElement) {
+        switch (headElement.type) {
+            case 'title':
+                setTitle(headElement.title);
+                break;
+            case 'tag':
+                const tag = createHeadTag(headElement);
+                prerenderedTags.push(tag);
+                break;
+        }
+    }
+
+    function parseHeadComment(node) {
+        if (!node || node.nodeType != Node.COMMENT_NODE) {
+            return;
+        }
+
+        const commentText = node.textContent;
+
+        if (!commentText) {
+            return;
+        }
+
+        const headStartComment = new RegExp(headCommentRegularExpression);
```

Comment:
Isn't `headCommentRegularExpression` already a regex?

Reply:
It is, but this was an extra precaution since `RegExp.exec()` is stateful [under certain conditions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec). Those conditions don't currently seem to meet in this case though so I can probably remove `headStartComment`.